### ISSUE TITLE
Overfitting during fine-tuning stage

### ISSUE THREAD

**Issue by DSCmatter** [SujalJariha/minor_project#6]
The training curves show clear overfitting after fine-tuning begins. Training loss drops to near 0 while validation loss remains at 0.02-0.05 with noisy oscillations, indicating the model is memorizing training data rather than generalizing.

## Evidence from training plots:

- Post fine-tuning: ~0.002 train loss vs ~0.02-0.05 val loss
- Validation loss spikes at epoch 10 when full model is unfrozen
- Val accuracy oscillates throughout, suggesting unstable validation signal

## Likely causes:

- Full ResNet50 unfreeze is too aggressive, too many layers updating at once
- Val set may be too small to give a stable signal
- Possible data leakage if augmented images weren't split before augmentation

## Suggested fixes:

- Unfreeze only the last 20-30 layers of ResNet50 instead of the full model
- Add Dropout(0.4) before the final dense layer
- Tighten early stopping patience
- Verify train/val split was done on original 850 images before augmentation

The graphs verify this from [/notebooks/model_dryfruit.ipynb](https://github.com/SujalJariha/minor_project/blob/main/notebooks/model_dryfruit.ipynb).

<img width="1527" height="552" alt="Image" src="https://github.com/user-attachments/assets/d4ad787f-bf92-421a-89e9-41ff08828a76" />
